Fix and enhance elevation prompt logic

harryytm opened the pull request:
- Add missing options to allow user re-enable prompts in menu
- Elevation prompt would remember user choice on being silenced
- Automatically grant elevated privileges when elevation prompt being silenced and selected Yes

NoMoreFood commented:
> Just to share my take on modding the elevation prompt logic. Tested works as how I desired, but the config page layout don't looks so good, can't find other logical places to put those 3 checkboxes in, other than increase the height of the page or creating a new config page for them, but I have to learn how to do that first.
> 
> @NoMoreFood Would really like you to give feedback on this mod.

I will take a look this weekend.

NoMoreFood commented on windirstat/WinDirStat.cpp:
```diff
@@ -284,6 +284,15 @@ BOOL CDirStatApp::InitInstance()
+    // Silently restart elevated conditionally before any expensive initialization
+    if (IsElevationAvailable() && COptions::ElevateOnDemand && !COptions::ShowElevationPrompt // only if user doesn't want to be prompted
+        // Any of these options enabled in settings would trigger elevation
+        && (COptions::UseFastScanEngine || COptions::UseBackupRestore))
```
I think I mentioned this before, but there are other reasons to run elevated even if you don't have these available so I would remove them from the logic.  If the user can elevate and they have opted to before, just do it.

NoMoreFood commented:
> Just to share my take on modding the elevation prompt logic. Tested works as how I desired, but the config page layout don't looks so good, can't find other logical places to put those 3 checkboxes in, other than increase the height of the page or creating a new config page for them, but I have to learn how to do that first.
> 
> @NoMoreFood Would really like you to give feedback on this mod.

I don't mind making it bigger, but maybe we should just make another button that controls all of the dialog prompts and call it "Reset Dialog Prompt Memory"?  That would also reduce executable bloat due to localization.

harryytm commented on windirstat/res/langs/lang_en.txt:
```diff
@@ -172,9 +172,12 @@ IDS_PAGE_GENERAL_FULL_ROW=&Full Row Selection
 IDS_PAGE_GENERAL_SHOW_GRID=Show &Grid
 IDS_PAGE_GENERAL_SHOW_SIZES=&Show sizes in GiB/MiB/KiB/bytes format
 IDS_PAGE_GENERAL_SHOW_STRIPES=Show S&tripes
+IDS_PAGE_GENERAL_START_ELEVATED=Automatically grant elevated privileges based on demand
```
Yes, if "If the user can elevate and they have opted to before, just do it." is what you mean.

harryytm commented:
<img width="348" height="275" alt="image" src="https://github.com/user-attachments/assets/0520759e-bdf3-43c5-b0d9-1efb9b3c29b8" />

Moved all 3 into Options and added some refined logic.

NoMoreFood commented:
Yeah, on second thought I'm regretting putting this under View.  I'm thinking about a more elegant way to handle it.

harryytm commented:
Added translations for all languages.

harryytm commented:
Changed "Automatically elevate if available" to "Automatically elevate on startup" in all languages translation which is more precise.
